pkg/identity/cache/allocator.go

```diff
@@ -482,3 +486,49 @@ func (m *CachingIdentityAllocator) RemoveRemoteIdentities(name string) {
 		m.IdentityAllocator.RemoveRemoteKVStore(name)
 	}
 }
+
+type IdentityChangeKind string
+
+const (
+	IdentityChangeSync   IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeSync)
+	IdentityChangeUpsert IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeUpsert)
+	IdentityChangeDelete IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeDelete)
+)
+
+type IdentityChange struct {
+	Kind   IdentityChangeKind
+	ID     identity.NumericIdentity
+	Labels labels.Labels
+}
+
+// Observe the identity changes. Conforms to stream.Observable.
+// Replays the current state of the cache when subscribing.
+func (m *CachingIdentityAllocator) Observe(ctx context.Context, next func(IdentityChange), complete func(error)) {
+	if m.IdentityAllocator != nil {
+		// Observe the underlying allocator for changes and map the events to
+		// identities.
+		m.IdentityAllocator.Observe(
```

Comment:
Nit: Consider using the `stream.Map` operator so you can reduce the boilerplate

Reply:
thanks @dylandreimerink . even though it doesn't remove that much boilerplate, it's way better to use APIs of package `stream` to unify such logic and the observable-patterns (instead of having to re-invent & document them)